frontend/src/e2e-test/specs/0_citizen/citizen-holiday-reservations.spec.ts

```diff
@@ -112,22 +150,65 @@ describe('Holiday periods', () => {
       await calendar.assertHolidayModalButtonVisible()
     })
 
-    test('Holidays can be reported', async () => {
+    test('Holidays can be reported and cleared', async () => {
+      const assertFreeAbsences = (hasFreeAbsences: boolean) =>
+        assertReservationState(
+          calendar,
+          LocalDate.of(2035, 12, 26),
+          LocalDate.of(2036, 1, 1),
+          hasFreeAbsences
+        )
+
       await enduserLogin(page)
       await new CitizenHeader(page).selectTab('calendar')
       const calendar = new CitizenCalendarPage(page, 'desktop')
 
-      await calendar.assertNoReservationsOrAbsences(LocalDate.of(2035, 12, 26))
+      await assertFreeAbsences(false)
 
-      const holidayModal = await calendar.openHolidayModal()
+      let holidayModal = await calendar.openHolidayModal()
```

Comment:
I don't quite get this pattern of returning page objects, because the page object do not hold any internal state.

> _Locator is resolved to the element immediately before performing an action, so a series of actions on the same locator can in fact be performed on different DOM elements. That would happen if the DOM structure between those actions has changed._

Using `let` and returning page objects leads to a false sense of immutability (or something) 😄

Reply:
I see it as a way to "mentally track" which page/modal is visible on the "screen". If an action opens a modal or moves to another page, then it returns the page model for that.